src/main/java/org/verdictdb/connection/InMemoryAggregate.java

```diff
@@ -0,0 +1,557 @@
+  private TreeMultiset<Pair<List<Object>, List<Object>>> rows;
+
+  private List<Object> currentRow;
+
+  private Iterator<Pair<List<Object>, List<Object>>> rowIterator;
+
+  private int cursor = -1;
+
+  // 0 is non-aggregate, 1 is sum or count, 2 is max, 3 is min
+  private List<Integer> isAggregate = new ArrayList<>();
+
+  private List<Integer> tierColumnIndex;
+
+  private List<String> aggColumnName;
+
+  private List<String> nonAggColumnName;
+
+  private List<String> columnName;
+
+  // 1 is sum or count, 2 is max, 3 is min
+  private List<Integer> isAggregateMaxMin = new ArrayList<>();
+
+  private Integer limit;
+
+  public InMemoryAggregate(SelectAsyncAggExecutionNode node) {
+    if (node.getSelectQuery().getLimit().isPresent()) {
+      ConstantColumn lim = (ConstantColumn) node.getSelectQuery().getLimit().get();
+      if (lim.getValue() instanceof String) {
+        limit = Integer.valueOf((String) lim.getValue());
+      } else {
+        limit = (Integer) lim.getValue();
+      }
+    }
+  }
+
+  public static void createTable(DbmsQueryResult dbmsQueryResult, String tableName) throws SQLException {
+    StringBuilder columnNames = new StringBuilder();
+    StringBuilder fieldNames = new StringBuilder();
+    StringBuilder bindVariables = new StringBuilder();
+    for (int i = 0; i < dbmsQueryResult.getColumnCount(); i++) {
+      if (i > 0) {
+        columnNames.append(", ");
+        fieldNames.append(", ");
+        bindVariables.append(", ");
+      }
+      fieldNames.append(dbmsQueryResult.getColumnName(i));
+      fieldNames.append(" ");
+      // char -> varchar in case this type is an array of char
+      int columnType = dbmsQueryResult.getColumnType(i);
+      if (columnType==CHAR) {
+        columnType=VARCHAR;
+      }
+      fieldNames.append(DataTypeConverter.typeName(columnType));
+      columnNames.append(dbmsQueryResult.getColumnName(i));
+      bindVariables.append('?');
+    }
+    // create table
+    String createSql = "CREATE TABLE IF NOT EXISTS " + tableName + " (" + fieldNames + ")";
+    conn.createStatement().execute(createSql);
+
+    // insert values
+    String sql = "INSERT INTO " + tableName + " ("
+        + columnNames
+        + ") VALUES ("
+        + bindVariables
+        + ")";
+    PreparedStatement statement = conn.prepareStatement(sql);
+    while (dbmsQueryResult.next()) {
+      for (int i = 1; i <= dbmsQueryResult.getColumnCount(); i++)
+        statement.setObject(i, dbmsQueryResult.getValue(i - 1));
+      statement.addBatch();
+    }
+    statement.executeBatch();
+  }
+
+  public static DbmsQueryResult executeQuery(SelectQuery query) throws VerdictDBException, SQLException {
+    String sql = selectQueryToSql.toSql(query).toUpperCase();
+    ResultSet rs = conn.createStatement().executeQuery(sql);
+    return new JdbcQueryResult(rs);
+  }
+
+  public static String combine(String combineTableName, String targetTableName, SelectQuery dependentQuery)
```

Comment:
If the first argument is the table that contains the values that have been aggregated thus far, let's use the name `combinedTableName`. Using a verb seems to be confusing.